### PR TITLE
Feature/appmesh servicediscovery changes

### PR DESCRIPTION
**Pull request by edugitforme**
New filter for appmesh resources like virtual service, virtual router, route and gateway route and new resource for service discovery namespace and filter for service discovery instance. Previously I added those filters as resources but than due to failure in test_policy.py I found that design wise, they are expected to work as filter to changed it.

### CODE REVIEW

File: c7n/resources/appmesh.py [hunk 1]
```diff
@@ -324,3 +327,125 @@ class resource_type(TypeInfo):
             'virtualNodes',
             None,
         )
+
+
+@AppmeshMesh.filter_registry.register('virtual-service')
```
Comment (kapilt) [inline c7n/resources/appmesh.py:332]: I think we can just substitute virtual in the resource name for "appmesh" so we can have these grouped together by the service from a naming perspective.

File: c7n/resources/appmesh.py [hunk 2]
```diff
@@ -324,3 +327,125 @@ class resource_type(TypeInfo):
...
+            response = self.manager.retry(client.list_virtual_services,
+                                          meshName=mesh["meshName"])
+
+            mesh[self.annotation_key] = response.get(
+                'virtualServices', [])
+
+        return super().process(resources, event)
+
+
+@AppmeshMesh.filter_registry.register('virtual-router')
```
Comment (kapilt) [inline c7n/resources/appmesh.py:361]: same with this one re s/virtual/appmesh

File: c7n/resources/appmesh.py [hunk 3]
```diff
@@ -324,3 +327,125 @@ class resource_type(TypeInfo):
...
+                response = self.manager.retry(client.list_routes,
+                                              meshName=virtual_router["meshName"],
+                                              virtualRouterName=virtual_router["virtualRouterName"])
+
+                virtual_router["routes"] = response.get("routes", [])
+
+        return super().process(resources, event)
+
+
+@AppmeshVirtualGateway.filter_registry.register('gateway-route')
```
Comment (kapilt) [inline c7n/resources/appmesh.py:424]: appmesh as resource prefix

File: c7n/resources/servicediscovery.py [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+
+from c7n.utils import (
+    local_session, type_schema
+)
+
+
+class DescribeServiceDiscoveryNamespace(DescribeSource):
+    # override default describe augment to get tags
+    def augment(self, resources):
+        detailed_resources = super(DescribeServiceDiscoveryNamespace, self).augment(resources)
```
Comment (kapilt) [inline c7n/resources/servicediscovery.py:22]: in python3, super calls can be simplified to 

`detailed_resources = super().augment(resources)`

as the language will automatically infer self and base class

File: c7n/resources/servicediscovery.py [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+        detail_spec = ('get_namespace', 'Id', 'Id', 'Namespace')
+
+        # refers to a field in the metadata response of the describe function
+        # https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/servicediscovery/client/get_namespace.html
+        date = 'CreateDate'
+
+
+@ServiceDiscoveryNamespace.filter_registry.register('servicediscovery-instance')
+class SdNamespaceInstance(ListItemFilter):
+
```
Comment (kapilt) [inline c7n/resources/servicediscovery.py:68]: can you add a doc string on example usage here, doc strings on classes and filters get automatically extracted for the reference documentation.

**Review by kapilt — COMMENTED** [review]
thanks for the pull request, this looks pretty good, some minors in comments.

File: c7n/resources/appmesh.py (hunk 1 above)
Comment (edugitforme) [reply c7n/resources/appmesh.py]: This is fixed to change it to appmesh-service. Please check

File: c7n/resources/appmesh.py (hunk 2 above)
Comment (edugitforme) [reply c7n/resources/appmesh.py]: This is fixed to change it to appmesh-router. Please check

File: c7n/resources/servicediscovery.py (hunk 4 above)
Comment (edugitforme) [reply c7n/resources/servicediscovery.py]: Done

File: c7n/resources/servicediscovery.py (hunk 5 above)
Comment (edugitforme) [reply c7n/resources/servicediscovery.py:68]: Added doc string on all resources

**Comment by edugitforme** [comment]
Thanks for the review @kapilt . I have fixed all comments and added doc string as well. All checks are passed. Kindly Please verify.

**Review by kapilt — COMMENTED** [review]
lgtm, sorry I think misunderstood these were filters when I suggested the prefix, I'm going to simplify as the resource already qualifies for filter vocabulary.

**Review by kapilt — APPROVED** [review]

**Pull request merged**